proxy/http2/Http2DependencyTree.h

```diff
@@ -257,17 +266,31 @@ void
 Http2DependencyTree<T>::_change_parent(Node *node, Node *new_parent, bool exclusive)
 {
   node->parent->children.remove(node);
+  if (node->queued) {
+    node->parent->queue->erase(node->entry);
```

Comment:
If the `queue` of parent become empty by this operation, the `queued` flag of parent should also be `false`.
We should do same operations to parent of parent recursively.
So I think we should walk up the tree to correct `queue` and `queued` flag of parents here. WDYT?